plugins/PaintTool/PaintTool.py

```diff
@@ -0,0 +1,286 @@
+                return self.undoStackAction(self._shift_held)
+            return False
+
+        if event.type == Event.MouseReleaseEvent and self._controller.getToolsEnabled():
+            if MouseEvent.LeftButton not in cast(MouseEvent, event).buttons:
+                return False
+            self._mouse_held = False
+            self._last_text_coords = None
+            return True
+
+        is_moved = event.type == Event.MouseMoveEvent
+        is_pressed = event.type == Event.MousePressEvent
+        if (is_moved or is_pressed) and self._controller.getToolsEnabled():
+            if is_moved and not self._mouse_held:
+                return False
+
+            evt = cast(MouseEvent, event)
+            if is_pressed:
+                if MouseEvent.LeftButton not in evt.buttons:
+                    return False
+                else:
+                    self._mouse_held = True
+
+            paintview = controller.getActiveView()
+            if paintview is None or paintview.getPluginId() != "PaintTool":
+                return False
+            paintview = cast(PaintView, paintview)
+
+            if not self._selection_pass:
+                return False
+
+            camera = self._controller.getScene().getActiveCamera()
+            if not camera:
+                return False
+
+            if node != self._node_cache:
+                if self._node_cache is not None:
+                    self._node_cache.transformationChanged.disconnect(self._nodeTransformChanged)
+                self._node_cache = node
+                self._node_cache.transformationChanged.connect(self._nodeTransformChanged)
+            if self._cache_dirty:
+                self._cache_dirty = False
+                self._mesh_transformed_cache = self._node_cache.getMeshDataTransformed()
+            if not self._mesh_transformed_cache:
+                return False
+
+            evt = cast(MouseEvent, event)
+
+            if not self._picking_pass:
+                self._picking_pass = PickingPass(camera.getViewportWidth(), camera.getViewportHeight())
+            self._picking_pass.render()
+
+            self._selection_pass.renderFacesMode()
+
+            texcoords = self._getTexCoordsFromClick(node, evt.x, evt.y)
+            if texcoords is None:
+                return False
+            if self._last_text_coords is None:
+                self._last_text_coords = texcoords
+
+            w, h = paintview.getUvTexDimensions()
+            sub_image, (start_x, start_y) = self._createStrokeImage(
+                self._last_text_coords[0] * w,
+                self._last_text_coords[1] * h,
+                texcoords[0] * w,
+                texcoords[1] * h
+            )
+            paintview.addStroke(sub_image, start_x, start_y)
+
+            self._last_text_coords = texcoords
+            Application.getInstance().getController().getScene().sceneChanged.emit(node)
+            return True
```

Comment:
For more clarity within this function, could you rename the various sub-casts of the event to more specific names, like `mouse_event` ?